src/Pages/Rooms/Rooms.tsx

```diff
@@ -0,0 +1,217 @@
+/** @format */
+
+import { NoImage5, deleteImg } from "@/Assets/Images";
+import { TableHeader } from "@/Components";
+import { RoomsData } from "@/Redux/Features/Rooms/GetRoomsSlice";
+import DeleteOutlineIcon from "@mui/icons-material/DeleteOutline";
+import EditIcon from "@mui/icons-material/Edit";
+import MoreVertIcon from "@mui/icons-material/MoreVert";
+import VisibilityIcon from "@mui/icons-material/Visibility";
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  ListItemIcon,
+  ListItemText,
+  MenuList,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+} from "@mui/material";
+import IconButton from "@mui/material/IconButton";
+import Menu from "@mui/material/Menu";
+import MenuItem from "@mui/material/MenuItem";
+import Slide from "@mui/material/Slide";
+import { indigo } from "@mui/material/colors";
+import React, { useCallback, useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+import "./Rooms.module.scss";
+import { deleteRoom } from "@/Redux/Features/Rooms/DeleteRoomSlice";
+
+import axios from "axios";
+import { useForm } from "react-hook-form";
+import DeleteModal from "@/Components/DeleteModal/DeleteModal";
+
+const Rooms = () => {
+  const [itemId, setItemId] = useState();
+  const dispatch = useDispatch();
+  const [tableData, setTableData] = useState(null);
+  const [openDialog, setOpenDialog] = useState(false);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+
+  {
+    /*Handle delete slice */
+  }
+  const deleteRecord = useCallback(
+    (el) => dispatch(deleteRoom(el.id)),
```

Comment:
having meaningful names is very crucial for better readability especially when working in a team, we can change `el` to `room`